### PR TITLE
Refactor vendors/smartfridges to reduce sprites, and fix existing sprites

### PR DESCRIPTION
**Pull request by Vi3trice**
## What Does This PR Do
Refactor the vendors/smartfridges to reduce sprites, from 246 sprites down to 181. Due to the sheer amount of shared sprites, or otherwise shared differences, this defines new variables to use for common sprites (e.g. All drobes share the same panel instead of having identical individual sprites)

Vendor_Sprites_Old
Vendor_Sprites_New

At the same time, this also makes the virology smartfridge subtype use its unique sprite as it was in the file, but unused. The half-filled circuit smartfridge also got fixed as its sprite was essentially wrong.

Viro_NewCircuits_New

The dye generator isn't a vendor nor smartfridge, but draws from the same file so it got tweaked as well. It also now has a defined `integrity_failure` value, allowing it to be broken, and repaired. Currently it just hits 0 integrity and gets destroyed, disappearing without leaving a trace.

## Why It's Good For The Game
Less sprites, opens up some new possibilities down the line. Dye generator is more consistent with vendors in terms of integrity behavior.

## Images of changes
Vendor_OnVendor_Off
Vendor_BrokenVendor_Panel

## Changelog
:cl:
fix: Fixed circuit smartfridge half-filled sprite
fix: Virology smartfridge uses its own sprite variant
fix: Dye Generator has a broken state
imagedel: Deleted redundant vendor sprites
/:cl:

### CODE REVIEW

File: code/game/machinery/dye_generator.dm [hunk 1]
```diff
@@ -2,8 +2,9 @@
 	name = "Dye Generator"
 	icon = 'icons/obj/vending.dmi'
 	icon_state = "barbervend"
-	density = 1
-	anchored = 1
+	density = TRUE
+	anchored = TRUE
+	integrity_failure = 100
```
Comment (dearmochi) [inline code/game/machinery/dye_generator.dm:7]: Undocumented change

File: code/game/machinery/vending.dm [hunk 2]
```diff
@@ -27,7 +27,11 @@
 	var/icon_vend
 	/// Icon_state when denying access
 	var/icon_deny
-
+	/// Icons for shared sprites, or overlays that go over a base sprite. If empty, it defaults to the icon_state as the base name for other states.
```
Comment (dearmochi) [inline code/game/machinery/vending.dm:30]: `///` is to be used per variable, not a bunch of them. Refer to the [autodoc guide](https://github.com/ParadiseSS13/Paradise/blob/master/.github/AUTODOC_GUIDE.md).

File: code/game/machinery/vending.dm [hunk 3]
```diff
@@ -1543,24 +1538,23 @@
 	armor = list(melee = 50, bullet = 20, laser = 20, energy = 20, bomb = 0, bio = 0, rad = 0, fire = 100, acid = 70)
 	resistance_flags = FIRE_PROOF
 
-
 /obj/machinery/vending/engivend
-	name = "\improper Engi-Vend"
+	name = "\improper Engi_vend"
```
Comment (dearmochi) [inline code/game/machinery/vending.dm:1542]:
```suggestion
	name = "\improper Engi-Vend"
```

probably un-intended change.

File: code/game/machinery/vending.dm [hunk 4]
```diff
@@ -1934,11 +1937,12 @@
 	contraband = list(/obj/item/toy/figure/crew/chemist = 1)
 	refill_canister = /obj/item/vending_refill/chemdrobe
 
-
 /obj/machinery/vending/genedrobe
 	name = "GeneDrobe"
```
Comment (dearmochi) [inline code/game/machinery/vending.dm:1941]:
```suggestion
	name = "\improper GeneDrobe"
```

File: code/game/machinery/vending.dm [hunk 5]
```diff
@@ -1955,7 +1959,9 @@
 /obj/machinery/vending/scidrobe
 	name = "SciDrobe"
```
Comment (dearmochi) [inline code/game/machinery/vending.dm:1960]:
```suggestion
	name = "\improper SciDrobe"
```

File: code/game/machinery/vending.dm [hunk 6]
```diff
@@ -1977,7 +1983,9 @@
 /obj/machinery/vending/robodrobe
 	name = "RoboDrobe"
```
Comment (dearmochi) [inline code/game/machinery/vending.dm:1984]:
```suggestion
	name = "\improper RoboDrobe"
```

File: code/game/machinery/vending.dm [hunk 7]
```diff
@@ -1994,7 +2002,9 @@
 /obj/machinery/vending/engidrobe
 	name = "EngiDrobe"
```
Comment (dearmochi) [inline code/game/machinery/vending.dm:2003]:
```suggestion
	name = "\improper EngiDrobe"
```

File: code/game/machinery/vending.dm [hunk 8]
```diff
@@ -2022,7 +2032,9 @@
 /obj/machinery/vending/atmosdrobe
 	name = "AtmosDrobe"
```
Comment (dearmochi) [inline code/game/machinery/vending.dm:2033]:
```suggestion
	name = "\improper AtmosDrobe"
```

File: code/game/machinery/vending.dm [hunk 9]
```diff
@@ -2050,7 +2062,9 @@
 /obj/machinery/vending/cargodrobe
 	name = "CargoDrobe"
```
Comment (dearmochi) [inline code/game/machinery/vending.dm:2063]:
```suggestion
	name = "\improper CargoDrobe"
```

File: code/game/machinery/vending.dm [hunk 10]
```diff
@@ -2070,7 +2084,9 @@
 /obj/machinery/vending/chefdrobe
 	name = "ChefDrobe"
```
Comment (dearmochi) [inline code/game/machinery/vending.dm:2085]:
```suggestion
	name = "\improper ChefDrobe"
```

File: code/game/machinery/vending.dm [hunk 11]
```diff
@@ -2090,6 +2106,7 @@
 	name = "BarDrobe"
```
Comment (dearmochi) [inline code/game/machinery/vending.dm:2106]:
```suggestion
	name = "\improper BarDrobe"
```

File: code/game/machinery/vending.dm [hunk 12]
```diff
@@ -2108,6 +2125,7 @@
 	name = "HydroDrobe"
```
Comment (dearmochi) [inline code/game/machinery/vending.dm:2125]:
```suggestion
	name = "\improper HydroDrobe"
```

File: code/game/machinery/vending.dm [hunk 13]
```diff
@@ -1649,22 +1644,22 @@
...
 	ads_list = list("The healthiest!","Award-winning chocolate bars!","Mmm! So good!","Oh my god it's so juicy!","Have a snack.","Snacks are good for you!","Have some more Getmore!","Best quality snacks straight from mars.","We love chocolate!","Try our new jerky!")
 	icon_state = "snack"
+	icon_panel = "thin_vendor"
 	products = list(/obj/item/reagent_containers/food/snacks/chips =6,/obj/item/reagent_containers/food/snacks/sosjerky = 6,
 					/obj/item/reagent_containers/food/snacks/syndicake = 6, /obj/item/reagent_containers/food/snacks/cheesiehonkers = 6)
 
-
 //don't forget to change the refill size if you change the machine's contents!
 /obj/machinery/vending/clothing
 	name = "\improper  ClothesMate" //renamed to make the slogan rhyme
```
Comment (dearmochi) [inline code/game/machinery/vending.dm:1659]:
```suggestion
	name = "\improper ClothesMate" //renamed to make the slogan rhyme
```

**Review by dearmochi — CHANGES_REQUESTED** [review]

File: code/game/machinery/dye_generator.dm (hunk 1 above)
Comment (Vi3trice) [reply code/game/machinery/dye_generator.dm:7]: Updated the description. Basically, it'd just disappear and wouldn't switch to a broken state due to defaulting to 0. This brings it in line with vendors.

**Review by dearmochi — APPROVED** [review]

**Pull request merged**